Xamarin.Forms.Core/Markup/BindableObjectExtensions.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+
+namespace Xamarin.Forms.Markup
+{
+	public static class BindableObjectExtensions
+	{
+		const string bindingContextPath = Binding.SelfPath;
+
+		/// <summary>Bind to a specified property</summary>
+		public static TBindable Bind<TBindable>(
```

Comment:
I'd prefer this to be named `SetBinding`

Reply:
That would go against the aim of eliminating often repeated noise words from markup.
These methods are equivalent to property setters (as the markup indentation reflects), which also do not require the developer to include the word **Set**

`Bind` occurs **a lot** and it is concise.  E.g. compare:
```CSharp
new Image { }
           .Bind (nameof(vm.Icon)),

new Label { LineBreakMode = LineBreakMode.WordWrap }
           .Bind (nameof(vm.RegistrationPrompt)),
                            
new Label { Text = "REGISTRATION CODE" },
        
new XEntry { Placeholder = "123456", Keyboard = Keyboard.Numeric }
            .Bind (XEntry.DisplayCheckmarkProperty, nameof(vm.IsRegistrationCodeFormatValid))
            .Bind (XEntry.ReturnCommandProperty, nameof(vm.VerifyRegistrationCodeCommand))
            .Bind (nameof(vm.RegistrationCode))
```
to:
```CSharp
new Image { }
            .SetBinding (nameof(vm.Icon)),

new Label { LineBreakMode = LineBreakMode.WordWrap }
            .SetBinding (nameof(vm.RegistrationPrompt)),
                            
new Label { Text = "REGISTRATION CODE" },
        
new XEntry { Placeholder = "123456", Keyboard = Keyboard.Numeric }
            .SetBinding (XEntry.DisplayCheckmarkProperty, nameof(vm.IsRegistrationCodeFormatValid))
            .SetBinding (XEntry.ReturnCommandProperty, nameof(vm.VerifyRegistrationCodeCommand))
            .SetBinding (nameof(vm.RegistrationCode))
```

In addition to that, `Bind` is consistent with all the other helper methods that **Set** a property, e.g.:
```CSharp
new Label { LineBreakMode = LineBreakMode.WordWrap }
           .Row (Row.Prompt) .CenterExpandH () .TextCenterH ()
           .Bind (nameof(vm.RegistrationPrompt)),
```